repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -316,6 +361,18 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
 
     // Track that the user intended to commit the response.
     flushedBuffer = true
+    committed = true
+  }
+
+  /**
+    * @throws IllegalStateException when the wrapped response has already been committed
+    */
+  def uncommit(): Unit = {
```

Comment:
Is there any case where the underlying won't be commited, but someone tries to call this when in non-mutable mode? If so we should check that as well.